sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -79,6 +79,10 @@ singleTableSchema
     : colTypeList EOF
     ;
 
+singleInterval
+    : INTERVAL? (intervalValue intervalUnit)+ EOF
```

Comment:
This is the existing behavior in 3.0, but I wonder if we really want to allow `SELECT INTERVAL 'interval 1 year'`. cc @MaxGekk @maropu @dongjoon-hyun

Reply:
The SQL standard doesn't define interval strings in such format with the `'interval'` prefix (or I haven't found that). If we look at other DBMS, for instance PostgreSQL, it supports strings without the prefix only.

I think we could support only strings without the prefix since Spark 3.0 because we haven't support bulk saving interval strings so far (in JSON or CSV, for example).

Reply:
Thank you for pinging me, @cloud-fan . It must be a corner case which we overlooked before. +1 for banning the case `SELECT INTERVAL 'INTERVAL 1 year'` in Apache Spark 3.0. What we need are only `SELECT INTERVAL '1 year'` and `SELECT 'INTERVAL 1 year'`. In other words,
- `SELECT INTERVAL 'INTERVAL 1 year'` (X)
- `SELECT INTERVAL '1 year'` (O)
- `SELECT 'INTERVAL 1 year'` (O)

Reply:
The @dongjoon-hyun suggested one looks ok to me.

Reply:
I'll send a followup PR to change this behavior and update related tests. Thanks for joining the discussion!